app/Http/Controllers/ScoresController.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ *    Copyright 2015-2018 ppy Pty. Ltd.
+ *
+ *    This file is part of osu!web. osu!web is distributed with the hope of
+ *    attracting more community contributions to the core ecosystem of osu!.
+ *
+ *    osu!web is free software: you can redistribute it and/or modify
+ *    it under the terms of the Affero GNU General Public License version 3
+ *    as published by the Free Software Foundation.
+ *
+ *    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+ *    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *    See the GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace App\Http\Controllers;
+
+use App\Models\Beatmap;
+use App\Models\Score\Best\Model as ScoreBest;
+use Auth;
+use PDOException;
+
+class ScoresController extends Controller
+{
+    public function report($mode, $id)
```

Comment:
If score validates id and mode, can't this be merged to `Users#report`? Looks like it'll simplify some things like the form.

Reply:
`UsersController#report` intentionally checks if it should 404 or not similarly to `#show`; the `ReportForm` component intentionally doesn't contain any network submission logic and the state and `onSubmit` parts of `ReportUser` and `ReportScore` are planned to be migrated to be `Provider`-based where they'd share the same implementation